Library/Homebrew/formula_installer.rb

```diff
@@ -571,7 +571,7 @@ def expand_dependencies
       keep_build_test ||= dep.build? && !install_bottle_for?(dependent, build) &&
                           (formula.head? || !dependent.latest_version_installed?)
 
-      if dep.prune_from_option?(build) || ((dep.build? || dep.test?) && !keep_build_test)
+      if dep.prune_from_option?(build) || ((dep.build? || dep.test? || dep.fetch?) && !keep_build_test)
```

Comment:
Will need to add something `dep.fetch?` related to `keep_build_test` above

Reply:
Did you mean line 544 in `expand_requirements`? I just added the same logic as for the `expand_dependencies` function.

Reply:
I mean something between lines 569-571 in this file. `keep_build_test` is handling whether it's being kept because it's needed for either testing or build. It's not handling the fetch case (or named accordingly) for those lines currently.